Build failure: python3*Packages.pgpy

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

- `nix-build -A python313Packages.pgpy`
- `nix-build -A python312Packages.pgpy`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/300450435

### Relevant log output

```console
FAILED tests/test_05_actions.py::TestPGPKey_Management::test_add_subkey[RSAEncryptOrSign-ECDH-EllipticCurveOID.Brainpool_P256] - TypeError: Can't instantiate abstract class BrainpoolP256R1 without an implementation for abstract method 'group_order'
FAILED tests/test_05_actions.py::TestPGPKey_Management::test_add_subkey[DSA-ECDH-EllipticCurveOID.Brainpool_P256] - TypeError: Can't instantiate abstract class BrainpoolP256R1 without an implementation for abstract method 'group_order'
FAILED tests/test_05_actions.py::TestPGPKey_Management::test_add_subkey[ECDSA-ECDH-EllipticCurveOID.Brainpool_P256] - TypeError: Can't instantiate abstract class BrainpoolP256R1 without an implementation for abstract method 'group_order'
FAILED tests/test_05_actions.py::TestPGPKey_Management::test_add_subkey[EdDSA-ECDH-EllipticCurveOID.Brainpool_P256] - TypeError: Can't instantiate abstract class BrainpoolP256R1 without an implementation for abstract method 'group_order'
FAILED tests/test_05_actions.py::TestPGPKey_Management::test_revoke_subkey[RSAEncryptOrSign-ECDH-EllipticCurveOID.Brainpool_P256] - StopIteration
FAILED tests/test_05_actions.py::TestPGPKey_Management::test_revoke_subkey[DSA-ECDH-EllipticCurveOID.Brainpool_P256] - StopIteration
FAILED tests/test_05_actions.py::TestPGPKey_Management::test_revoke_subkey[ECDSA-ECDH-EllipticCurveOID.Brainpool_P256] - StopIteration
FAILED tests/test_05_actions.py::TestPGPKey_Management::test_revoke_subkey[EdDSA-ECDH-EllipticCurveOID.Brainpool_P256] - StopIteration
```

### Additional context

Assuming this is due to `cryptography` bump: https://github.com/NixOS/nixpkgs/commit/38c2714181efb47dd95d57baa45e84b951349be9

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.2, NixOS, 25.11 (Xantusia), 25.11pre816272.9e83b64f727c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/puna/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/puna/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/puna/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/puna/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/store/ra9zaqz1hmrh6lj9g9g424v0jnchxf5y-gnome-settings-daemon-48.1/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/3bpcfg9842vvcka9g4rb6ab600j0yxmj-lix-2.91.1/share`
 - channels(bt1cn): `"unstable"`
 - channels(root): `"nixos, nixos-25.05-25.05, unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers

@eadwu @dotlambda

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pgpy/Fix-compat-with-current-cryptography.patch
+++ b/pkgs/development/python-modules/pgpy/Fix-compat-with-current-cryptography.patch
@@ -1,0 +1,49 @@
+From 5cdda87f74bcbb1dd7d29bb49b6a0ee67e41a7ce Mon Sep 17 00:00:00 2001
+From: OPNA2608 <opna2608@protonmail.com>
+Date: Thu, 10 Jul 2025 17:51:02 +0200
+Subject: [PATCH] pgpy/_curves.py: Fix compat with current cryptography
+
+Curves must now specify their group_order.
+---
+ pgpy/_curves.py | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/pgpy/_curves.py b/pgpy/_curves.py
+index 14f2528..91e7dbb 100644
+--- a/pgpy/_curves.py
++++ b/pgpy/_curves.py
+@@ -75,26 +75,31 @@ else:
+     class BrainpoolP256R1(ec.EllipticCurve):
+         name = 'brainpoolP256r1'
+         key_size = 256
++        group_order = 0xa9fb57dba1eea9bc3e660a909d838d718c397aa3b561a6f7901e0e82974856a7
+ 
+ 
+     class BrainpoolP384R1(ec.EllipticCurve):  # noqa: E303
+         name = 'brainpoolP384r1'
+         key_size = 384
++        group_order = 0x8cb91e82a3386d280f5d6f7e50e641df152f7109ed5456b31f166e6cac0425a7cf3ab6af6b7fc3103b883202e9046565
+ 
+ 
+     class BrainpoolP512R1(ec.EllipticCurve):  # noqa: E303
+         name = 'brainpoolP512r1'
+         key_size = 512
++        group_order = 0xaadd9db8dbe9c48b3fd4e6ae33c9fc07cb308db3b3c9d20ed6639cca70330870553e5c414ca92619418661197fac10471db1d381085ddaddb58796829ca90069
+ 
+ 
+     class X25519(ec.EllipticCurve):  # noqa: E303
+         name = 'X25519'
+         key_size = 256
++        group_order = 0x1000000000000000000000000000000014def9dea2f79cd65812631a5cf5d3ed
+ 
+ 
+     class Ed25519(ec.EllipticCurve):  # noqa: E303
+         name = 'ed25519'
+         key_size = 256
++        group_order = 0x1000000000000000000000000000000014def9dea2f79cd65812631a5cf5d3ed
+ 
+ 
+ # add these curves to the _CURVE_TYPES list
+-- 
+2.49.0
+

--- a/pkgs/development/python-modules/pgpy/default.nix
+++ b/pkgs/development/python-modules/pgpy/default.nix
@@ -34,6 +34,9 @@ buildPythonPackage rec {
   patches = [
     # https://github.com/SecurityInnovation/PGPy/issues/462
     ./pr-443.patch
+
+    # https://github.com/SecurityInnovation/PGPy/pull/474
+    ./Fix-compat-with-current-cryptography.patch
   ];
 
   postPatch = ''

